### DISCUSSION TITLE
Iteration support?

### DISCUSSION THREAD

**Discussion by psmoot** [lonelyteapot/pygraphic#10] [Q&A]
I'm looking for a library to use when querying GitHub's GraphQL interface. 

One of the issues we will have is long lists. Those will require dealing with GraphQL pagination. I'd love to have an interface which handles the boilerplate of iterating through a long list until the end, giving us a single (huge) data structure as a result. Has there been any talk of adding de-pagination support to this pygraphic?

**Comment by lonelyteapot** [comment]
I'll think about this. Thanks for the idea!

This data structure should probably be implemented like an indexed array with cache.

If you could provide an example on how it would be used, please do
